config/config.go

```diff
@@ -989,6 +998,24 @@ func SetupViper(v *viper.Viper, filename string) {
 	v.SetEnvPrefix("PBS")
 	v.AutomaticEnv()
 	v.ReadInConfig()
+
+	// Migrate config settings to maintain compatibility with old configs
+	migrateConfig(v)
+}
+
+func migrateConfig(v *viper.Viper) {
+	//v.RegisterAlias("stored_requests.directorypath", "stored_requests.filesystem.directorypath")
+
+	// if stored_requests.filesystem is boolean in conf file, migrate to new struct to avoid breaking viper
+	switch v.Get("stored_requests.filesystem").(type) {
+	case bool:
+		glog.Warning("stored_requests.filesystem should be changed to stored_requests.filesystem.enabled")
+		glog.Warning("stored_requests.directorypath should be changed to stored_requests.filesystem.directorypath")
+		m := v.GetStringMap("stored_requests.filesystem")
```

Comment:
Did you test if this works with the environment variable configs too?

Reply:
Good catch. It fails miserably with `PBS_STORED_REQUESTS_FILESYSTEM_ENABLED`.